packages/core/custom-branding/core-custom-branding-browser-internal/src/custom_branding_service.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { BehaviorSubject, Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+import type { CustomBrandingStart, CustomBrandingSetup } from '@kbn/core-custom-branding-browser';
+import type { CustomBranding } from '@kbn/core-custom-branding-common';
+
+const CUSTOM_BRANDING_PLUGIN = 'customBranding';
+
+export class CustomBrandingService {
+  private customBranding: CustomBranding = {};
+  private registeredPlugin?: string;
+  private hasCustomBranding$: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);
+  private customBranding$: BehaviorSubject<CustomBranding> = new BehaviorSubject<CustomBranding>(
+    this.customBranding
+  );
+  private stop$ = new Subject<void>();
+
+  private set(customBranding: CustomBranding) {
+    if (this.registeredPlugin !== CUSTOM_BRANDING_PLUGIN) {
```

Comment:
Why not do this check on L42 if there is only one plugin name allowed? And here validate that it is not undefined?